File: rust-typescript/project/lineReader.ts
// Run: npx ts-node ./lineReader.ts

import fs from "fs";

// Print every line from file
fs.readFileSync("./lines")
  .toString()
  .split("\n")
  .forEach((line) => console.log(line));

// Print every other line
fs.readFileSync("./lines")
  .toString()
  .split("\n")
  .filter((_, i) => i % 2 === 0)
  .forEach((line) => console.log(line));

File: rust-typescript/project/line_reader.rs
// Build: rustc line_reader.rs
// Run ./line_reader

fn main() {
    // Print every line from file
    let file = std::fs::read_to_string("./lines").unwrap();
    file.lines().for_each(|line| println!("{}", line));

    // Print every other line
    let file = std::fs::read_to_string("./lines").unwrap();
    file.lines()
        .enumerate()
        .filter(|(i, _)| return i % 2 == 0)
        .for_each(|(_, line)| println!("{}", line));
}
